hack/generate-swagger-api.sh

```diff
@@ -1,65 +1,96 @@
-#!/bin/sh
+#!/bin/bash
+# vim: set noexpandtab:
+# -*- indent-tabs-mode: t -*-
 set -eu
 
-swagger generate model -f api/swagger.yaml \
-	-t api -m types/plugin -C api/swagger-gen.yaml \
-	-T api/templates --allow-template-override \
-	-n Plugin \
-	-n PluginDevice \
-	-n PluginMount \
-	-n PluginEnv
-
-swagger generate model -f api/swagger.yaml \
-	-t api -m types/common -C api/swagger-gen.yaml \
-	-T api/templates --allow-template-override \
-	-n IDResponse \
-	-n ErrorResponse
-
-swagger generate model -f api/swagger.yaml \
-	-t api -m types/storage -C api/swagger-gen.yaml \
-	-T api/templates --allow-template-override \
-	-n DriverData
-
-swagger generate model -f api/swagger.yaml \
-	-t api -m types/container -C api/swagger-gen.yaml \
-	-T api/templates --allow-template-override \
-	-n ContainerCreateResponse \
-	-n ContainerUpdateResponse \
-	-n ContainerTopResponse \
-	-n ContainerWaitResponse \
-	-n ContainerWaitExitError \
-	-n ChangeType \
-	-n FilesystemChange \
-	-n PortSummary
-
-swagger generate model -f api/swagger.yaml \
-	-t api -m types/image -C api/swagger-gen.yaml \
-	-T api/templates --allow-template-override \
-	-n ImageDeleteResponseItem
-#-n ImageSummary TODO: Restore when go-swagger is updated
+generate_model() {
+	local package="$1"
+	mapfile
+	swagger generate model --spec=api/swagger.yaml \
+		--target=api --model-package="$package" \
+		--config-file=api/swagger-gen.yaml \
+		--template-dir=api/templates --allow-template-override \
+		$(printf -- '--name=%s ' "${MAPFILE[@]}")
+}
+
+generate_operation() {
+	mapfile
+	swagger generate operation --spec=api/swagger.yaml \
+		--target=api --api-package=types --model-package=types \
+		--config-file=api/swagger-gen.yaml \
+		--template-dir=api/templates --allow-template-override \
+		"$@" \
+		$(printf -- '--name=%s ' "${MAPFILE[@]}")
+}
+
+# /==================================================================\
+# |                                                                  |
+# |  ATTENTION:                                                      |
+# |                                                                  |
+# |       Sort model package stanzas and model/operation names       |
+# |                    *** ALPHABETICALLY ***                        |
+# |          to reduce the likelihood of merge conflicts.            |
+# |                                                                  |
+# \==================================================================/
+
+#region -------- Models --------
+
+generate_model types/common <<-'EOT'
+	ErrorResponse
+	IDResponse
+EOT
+
+generate_model types/container <<-'EOT'
```

Comment:
ISTR that in one of your PRs you (temporarily) added the package name in the swagger file; I removed that, but later considered; what if we added the correct package name in all of them; that way all models could be generated in one go. Not sure if a good idea though.

Reply:
Are you referring to commits ee560a3b23191e3e37838128a1cf3d7db29b489c and c13266d2c0abdf19bcaa4ad11d309a09601d0681? The package name you're talking about is a hint to go-swagger as to the import path _of a referenced external type._ You were able to remove that hint [because go-swagger defaults to the "target models defined by the `--model-package` flag".](https://goswagger.io/go-swagger/reference/models/schemas/#external-type-hints) and you moved the generated code into the same package as the referenced hand-written external types.

Reply:
Ah! Probably that was it, yes, and maybe I _thought_ that same option could be used for the type itself, but definitely didn't try!